peachjam/js/components/report_document_issue.ts

```diff
@@ -0,0 +1,39 @@
+export class ReportDocumentIssue {
+  root: HTMLElement;
+  form: HTMLFormElement | null;
+
+  constructor (root: HTMLElement) {
+    this.root = root;
+    this.form = this.root.querySelector('form');
+    if (this.form) {
+      this.form.addEventListener('submit', (e: SubmitEvent) => this.submit(e));
+    }
+  }
+
+  submit (e: SubmitEvent) {
+    e.preventDefault();
+
+    const form = new FormData(this.form || undefined);
+
+    fetch('/api/document-problem/', {
+      method: 'post',
+      body: form
+    }).then(response => {
+      if (response.ok) {
+        return response.text();
+      } else {
+        throw new Error('Something went wrong');
```

Comment:
Since this form can handle an error, how about `return "Something went wrong"` so that it's put into the body of the modal?